lib/puppet/util/servicenow.rb

```diff
@@ -71,13 +71,18 @@ def settings(settings_file = Puppet[:confdir] + '/servicenow_reporting.yaml')
   end
   module_function :settings
 
-  def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token: nil)
+  def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token: nil, skip_cert_check: false)
     uri = URI.parse(uri)
 
+    opts = {
+      use_ssl: uri.scheme == 'https',
+    }
+
+    opts[:verify_mode] = OpenSSL::SSL::VERIFY_NONE if skip_cert_check
+
     Net::HTTP.start(uri.host,
                     uri.port,
-                    use_ssl: uri.scheme == 'https',
-                    verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+                    opts) do |http|
```

Comment:
Just a heads up that puppetserver in fips mode can't load `net/http`, which would affect the snow report processor. In Puppet 5/6, you can call https://github.com/puppetlabs/puppet/blob/fe73adb22453824c014d7975e30e4fc882e8bbc2/lib/puppet/network/http_pool.rb#L36 and puppet and puppetserver will "do the right thing". /cc @justinstoller

Reply:
I think switching from Net::HTTP to is going to be quite a lot of changes that aren't specifically related to just enabling SSL in the current code. I think I'll close this ticket and PR and raise another one that's titled more appropriately for making the switch to a new method of doing HTTP calls. Thanks for your help.